src/content/articles/why-typescript-doesnt-include-a-throws-keyword.mdx

```diff
@@ -0,0 +1,370 @@
+Languages such as Java allow adding a type annotation alongside caught exceptions to run logic specific
+
+This hypothetical Java code runs specific logic for caught `ValueException`s, and falls back to more general logic for other `Exception`s:
+
+```java
+public void example() {
+  try {
+    Assert.positive(-1);
+  } catch (ValueException error) {
+    System.out.println("Incorrect value: " + error.value);
+  } catch (Exception error) {
+    System.out.println("General error: " + error.message);
+  }
+}
+```
+
+Checked exceptions are a handy tool for running different logic based on the type of a thrown exception.
+Strongly typed languages like Java are able to enforce the correct `catch` block is run based on the type of the caught exception.
+
+## Barriers to Throw Types
+
+In practice, there are quite a few reasons why throw types aren't feasible in the TypeScript language.
+These ranging from what's practically possible given common JavaScript practices up through difficulties of truly representing throw types in the type system.
+
+### Ubiquitous Unchecked Untyped Exceptions
+
+It is an unfortunate reality in coding that most lines of code can throw all sorts of errors unexpectedly.
+Even seemingly type-safe code can sometimes mysteriously throw an error, including Object getters and setters.
+
+For example, setting [the `length` of an `Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length) to a negative or too-large will throw a [`RangeError`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError):
+
+```ts twoslash
+[].length = -1;
+// Runtime error thrown: "RangeError: Invalid array length"
+```
+
+User-defined getters and setters may throw errors too.
+The following `Counter` class intentionally is able to throw errors in its `count` property's getter:
+
+```ts twoslash
+class Counter {
+  #counted = 0;
+
+  get count() {
+    if (!this.#counted) {
+      throw new Error("Not ready yet.");
+    }
+
+    return this.#counted;
+  }
+
+  increment() {
+    this.#counted += 1;
+  }
+}
+
+const { count } = new Counter();
+// Runtime error thrown: "Error: Not ready yet."
+```
+
+Even worse, JavaScript doesn't guarantee thrown objects to be instances of its built-in `Error` class!
+The following code, horrifyingly, throws one of four types, two of which are not `Error` instances:
+
+```ts twoslash
+function thisIsValidTypeScript() {
+  switch (Math.floor(Math.random() * 4)) {
+    case 0:
+      throw new RangeError("Zero?!");
+    case 1:
+      throw new Error("Gotcha!");
+    case 2:
+      throw "a primitive string, not an Error";
+    case 3:
+      throw null;
+  }
+}
+```
+
+:::tip
+The [`plugin:@typescript-eslint/only-throw-error`](https://typescript-eslint.io/rules/only-throw-error) lint rule can enforce writing code that only ever throws `Error`s.
```

Comment:
Just noting that this sentence is blocked on https://github.com/typescript-eslint/typescript-eslint/pull/8701. I'd really like to get that PR in before this blog post releases.